### PR TITLE
[searchcache] Triaged Search Atom 

### PR DESCRIPTION
**Pull request by KyleJu**
Fix for #1760

### CODE REVIEW

**Comment by KyleJu** [comment]
Ready for review. Will add the code for wildcards once https://github.com/web-platform-tests/wpt.fyi/pull/1765 is submitted

File: webapp/components/test/test-search.html [hunk 1]
```diff
@@ -396,6 +396,19 @@
       });
     });
 
+    test('simple triaged search', () => {
+      assertQueryParse('triaged:chrome', { exists: [{ triaged: 'chrome' }] });
+    });
+
+    test('test status | triaged', () => {
```
Comment (stephenmcgruer) [inline webapp/components/test/test-search.html]: Isn't this an 'and', not an 'or'? Perhaps I misunderstand what you meant by 'test status | triaged' - I suggest giving tests more descriptive names.

**Comment by stephenmcgruer** [comment]
Oh, the query worked the second time around. Sigh.

File: api/query/atoms.go [hunk 2]
```diff
@@ -252,6 +252,34 @@ func (l AbstractLink) BindToRuns(runs ...shared.TestRun) ConcreteQuery {
...
+// BindToRuns for AbstractTriaged binds each run of the AbstractTriaged ProductSpec
+// to a Traiged object.
+func (t AbstractTriaged) BindToRuns(runs ...shared.TestRun) ConcreteQuery {
+	cq := make([]ConcreteQuery, 0)
+	var netClient = &http.Client{
+		Timeout: time.Second * 5,
+	}
+	for _, run := range runs {
+		if t.Product == nil || t.Product.Matches(run) {
+			metadata, _ := shared.GetMetadataResponse([]shared.TestRun{run}, netClient, logrus.StandardLogger(), shared.MetadataArchiveURL)
```
[the remark this answers is not in the document]
Comment (KyleJu) [reply api/query/atoms.go]: Correct. I will fix https://github.com/web-platform-tests/wpt.fyi/issues/1776 before landing this PR

File: api/query/atoms.go [hunk 3]
```diff
@@ -1025,5 +1085,11 @@ func unmarshalQ(b []byte) (AbstractQuery, error) {
 			return i, nil
 		}
 	}
+	{
+		var t AbstractTriaged
+		if err := json.Unmarshal(b, &t); err == nil {
+			return t, nil
+		}
+	}
 	return nil, errors.New(`Failed to parse query fragment as test name pattern, test status constraint, negation, disjunction, conjunction, sequential or count`)
```
[the remark this answers is not in the document]
Comment (KyleJu) [reply api/query/atoms.go]: Yea its only gonna get longer

File: api/query/cache/index/filter.go [hunk 4]
```diff
@@ -303,9 +310,50 @@ func (l Link) Filter(t TestID) bool {
 	return false
 }
 
+// Filter interprets a Traiged as a filter function over TestIDs.
```
[the remark this answers is not in the document]
Comment (KyleJu) [reply api/query/cache/index/filter.go]: Good catch!

File: api/query/cache/index/filter.go [hunk 5]
```diff
@@ -303,9 +310,50 @@ func (l Link) Filter(t TestID) bool {
...
+	if !ok {
+		return false
+	}
+
+	if len(val) == 0 {
+		return false
+	}
+
+	if len(val) > 1 {
+		logrus.Errorf("Error executing filter query %v: test %v has more than one url", tr, name)
```
[the remark this answers is not in the document]
Comment (KyleJu) [reply api/query/cache/index/filter.go]: Correct. IIUC, if you look at the `BindToRuns` method, each  Triaged atom is bound to one test run that matches the productSpec of the atom.

**Comment by KyleJu** [comment]
This PR is blocked by https://github.com/web-platform-tests/wpt.fyi/issues/1776

File: api/query/atoms.go [hunk 6]
```diff
@@ -252,6 +252,34 @@ func (l AbstractLink) BindToRuns(runs ...shared.TestRun) ConcreteQuery {
...
+// BindToRuns for AbstractTriaged binds each run of the AbstractTriaged ProductSpec
+// to a triaged object.
+func (t AbstractTriaged) BindToRuns(runs ...shared.TestRun) ConcreteQuery {
+	cq := make([]ConcreteQuery, 0)
+	var netClient = &http.Client{
+		Timeout: time.Second * 5,
+	}
+	for _, run := range runs {
+		if t.Product == nil || t.Product.Matches(run) {
+			metadata, _ := shared.GetMetadataResponse([]shared.TestRun{run}, netClient, logrus.StandardLogger(), shared.MetadataArchiveURL)
```
Comment (KyleJu) [inline api/query/atoms.go]: Note to myself: API for this method has changed in the caching PR

**Pull request merged**